File: app/src/main/java/com/domclick/dto/AccountDto.java
package com.domclick.dto;

import com.domclick.model.Account;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

public class AccountDto implements Serializable {
    private Long id;
    private BigDecimal balance;
    private List<LinkDto> links = new ArrayList<>();

    public AccountDto(Account account) {
        this.id = account.getId();
        this.balance = account.getBalance();
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public BigDecimal getBalance() {
        return balance;
    }

    public void setBalance(BigDecimal balance) {
        this.balance = balance;
    }

    public List<LinkDto> getLinks() {
        return links;
    }

    public void setLinks(List<LinkDto> links) {
        this.links = links;
    }
}

File: app/src/main/kotlin/com/domclick/dto/AccountDto.kt
package com.domclick.dto

import com.domclick.model.Account
import lombok.Builder

import java.io.Serializable
import java.math.BigDecimal
import java.util.ArrayList

@Builder
class AccountDto(account: Account) : Serializable {
    var id: Long? = null
    var balance: BigDecimal? = null
    var links: ArrayList<LinkDto> = ArrayList()

    init {
        this.id = account.id
        this.balance = account.balance
    }
}
